### PR TITLE
Fix blank screen when showing the WorkspaceResetBankAccount modal

### PR DESCRIPTION
**Pull request by MariaHCD**
### Details
Because we return the WorkspaceResentBankAccountModal [here](https://github.com/Expensify/App/blob/e1c554a1f5dafe3afc787a069001d55372ce98ef/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L266-L270), we're inadvertently hiding the [Continue with setup](https://github.com/Expensify/App/blob/e1c554a1f5dafe3afc787a069001d55372ce98ef/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L272-L290) form leading to a blank screen. Similarly, we hide the EnableStep and ValidationStep components.

This PR ensure that the ContinueBankAccountSetup, EnableStep and ValidationStep are still visible when the WorkspaceResetBankAccount modal is shown.

### Fixed Issues

$ https://github.com/Expensify/App/issues/14692
$ https://github.com/Expensify/App/issues/14566   
PROPOSAL: NA

#### Bank account in setup
1. Navigate to Workspace > Connect bank account
2. Start the process of adding a VBA 
3. Exit the flow midway
4. Navigate to Workspace > Connect bank account again
5. Once you see Continue with setup, click `Start over`
6. Verify the modal is shown and Continue with step is still shown in the background

Screenshot 2023-02-01 at 1 33 05 PM

7. Verify that clicking `Cancel` or the confirm button in the modal works as expected

#### Verifying bank account
1. Follow [this SO](https://stackoverflowteams.com/c/expensify/questions/342) to create a VERIFYING bank account. If you don't have access to the StackOverflow use the following details:

```
Select "Manually enter account details" and enter Routing number 011401533 and Account number 1111222233331111
Company Info - enter random info but complete all fields. You will fail once and get asked to double check information. Select "Save & Continue" again
Personal Info - use any test data. For last 4 SSN use 3333.
Onfido - add a test ID
Beneficial Owners:
- Toggle terms and conditions
- Toggle confirmation that the information provided is true and accurate
```
2. Verify you see the ValidationStep
3. Click `No, start over` and verify that the confirm modal is shown along with the ValidationStep in the background

Screenshot 2023-02-01 at 1 50 53 PM

4. Verify that clicking `Cancel` or the confirm button in the modal works as expected

#### Open bank account
1. Follow [this SO](https://stackoverflowteams.com/c/expensify/questions/342) to create an OPEN bank account. If you do not have access to the StackOverflow, follow the below steps:

```
Click "Log into your bank"
- Plaid modal - Choose Wells Fargo (use Bank of America if Wells Fargo is not available)
- Plaid modal - User Id: user_good
- Plaid modal - Password: pass_good
- Plaid modal - Submit
- Plaid modal - If they request a verification code via mobile sms, enter: credential_good when required
- Choose the Saving account: Plaid Saving11122XXXXXX111
- Save & Continue

Company Information (Enter the info below, and anything in non-specified fields):
- Legal Business Name: Alberta Bobbeth Charleson
- Company website: Enter any valid formatted website
- Tax ID number: 123456789
- Confirm company is not on list of restricted businesses

Requestor Information (Enter the info below, and anything in non-specified fields)
- First name: Alberta
- Second name: Charleson
- Toggle confirmation you are authorized to use company bank account
- Save & Continue (Note: You will receive an error asking to "Please verify you name and date of birth". Hit "Save & Continue" again without changing any inputs.)

Beneficial Owners
- Toggle terms and conditions
- Toggle confirmation that the information provided is true and accurate
```
2. Verify that you see the EnableStep
3. Click `Disconnect bank account` and verify that the confirm modal is shown along with the EnableStep in the background

Screenshot 2023-02-01 at 1 34 53 PM

4. Verify that clicking `Cancel` or the confirm button in the modal works as expected

### Offline tests

1. Go offline on the Continue with setup page
2. Verify that clicking Start over shows the confirmation modal and the background screen does not go blank
3. Verify that clicking cancel dismisses the confirmation modal
4. Verify that confirming the modal leads you to the BankAccount step

### QA Steps

Same as test steps

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Open bank account.webm](https://user-images.githubusercontent.com/12268372/216009589-f9cef1d6-f196-49ae-9176-73b88d45b132.webm)

[Verifying bank account.webm](https://user-images.githubusercontent.com/12268372/216009592-09f3b6ab-b8b9-4e08-8e98-345043d91a9b.webm)

[Bank account in setup.webm](https://user-images.githubusercontent.com/12268372/216009579-89efbd30-d2a9-4068-969a-c6467c69d373.webm)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot_20230201-144451

Screenshot_20230201-144432

</details>

<details>
<summary>Mobile Web - Safari</summary>

Simulator Screen Shot - iPhone 14 - 2023-02-01 at 14 43 30
Simulator Screen Shot - iPhone 14 - 2023-02-01 at 14 43 06

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-02-02 at 1 25 01 PM

Screenshot 2023-02-02 at 1 22 56 PM

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-02-09 at 3 32 16 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-02-09 at 3 46 26 PM

</details>

### CODE REVIEW

File: src/pages/workspace/WorkspaceResetBankAccountModal.js [hunk 1]
```diff
@@ -12,9 +13,16 @@ const propTypes = {
...
     reimbursementAccount: ReimbursementAccountProps.reimbursementAccountPropTypes.isRequired,
 
+    /** Callback when the user confirms resetting the workspace bank account */
+    onConfirm: PropTypes.func,
+
     ...withLocalizePropTypes,
 };
 
+const defaultProps = {
+    onConfirm: null,
```
Comment (nkuoch) [inline src/pages/workspace/WorkspaceResetBankAccountModal.js:23]: or you can default to `() => {}` and then no need for `if (props.onConfirm) {` below

**Comment by MariaHCD** [comment]
Updated! 

One thing I noticed when testing (and I'm not sure what the cause is) but if you click start over and then quickly click `Connect with Plaid` or `Connect manually`, it takes you to the next page but then quickly reverts back to the previous page with the options:

[(1) New Expensify.webm](https://user-images.githubusercontent.com/12268372/216557106-3db25a65-c859-4217-b633-96dbba51d3d0.webm)

**Comment by MariaHCD** [comment]
Ah, I think the above error is because the [subStep](https://github.com/Expensify/App/blob/982b971b5fd31bb5f57a0204c587feb84eec56c7/src/pages/ReimbursementAccount/BankAccountStep.js#L132-L134) is being overridden and set null when [resetFreePlanBankAccount completes ](https://github.com/Expensify/App/blob/maria-workspace-bank-account-reset-fix/src/libs/actions/ReimbursementAccount/resetFreePlanBankAccount.js#L42-L46)

**Comment by MariaHCD** [comment]
Fixed it by adding `isLoading` to the Onyx data for `RestartBankAccountSetup` so we have a loading screen until the API command is complete:

[(1) New Expensify (1).webm](https://user-images.githubusercontent.com/12268372/216571782-37fd4f58-4b94-42d9-aa85-ac115f951d29.webm)

File: src/libs/actions/ReimbursementAccount/resetFreePlanBankAccount.js [hunk 2]
```diff
@@ -44,12 +44,31 @@ function resetFreePlanBankAccount(bankAccountID) {
...
+                    value: {isLoading: false},
+                },
+            ],
+            failureData: [
+                {
+                    onyxMethod: CONST.ONYX.METHOD.MERGE,
+                    key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
+                    value: {isLoading: false},
+                },
+            ]
```
Comment (nkuoch) [inline src/libs/actions/ReimbursementAccount/resetFreePlanBankAccount.js:71]: Missing comma

File: src/pages/ReimbursementAccount/ContinueBankAccountSetup.js [hunk 3]
```diff
@@ -67,13 +73,22 @@ const ContinueBankAccountSetup = props => (
                     <MenuItem
                         title={props.translate('workspace.bankAccount.startOver')}
                         icon={Expensicons.RotateLeft}
-                        onPress={props.startOver}
+                        onPress={() => {
```
Comment (nkuoch) [inline src/pages/ReimbursementAccount/ContinueBankAccountSetup.js:76]: can be on same line

**Review by nkuoch — DISMISSED** [review]

**Review by rushatgabhane — COMMENTED** [review]
## Reviewer Checklist



- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Disconnect

https://user-images.githubusercontent.com/29673073/216710600-7f691cc6-b3ea-497c-92a5-488daff8fbbc.mov


Reset progress

https://user-images.githubusercontent.com/29673073/216619939-e6e5e31c-0f00-4eb0-a26d-2256dac72aef.mov


Validate

https://user-images.githubusercontent.com/29673073/216710558-cb47e769-0bc7-4374-a400-6fc894b5d2ca.mov
</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/29673073/218459071-0c184825-24e2-4883-8bc1-7059fad1b71a.mp4



</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/29673073/218457564-7683da4b-31cf-4856-baa6-dcfea2c91930.mp4



</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/29673073/216711153-5e19ddf7-4a4e-43cf-b5e0-1839d0d08e14.mov


https://user-images.githubusercontent.com/29673073/216711491-ee51f406-589d-4530-951f-b06fc710dff2.mov



</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/29673073/218455662-e2bb07b9-6cdc-4bfe-bad3-e8e6d48ea885.mp4



</details>

<details>
<summary>Android</summary>


https://user-images.githubusercontent.com/29673073/218459111-79bd3cc9-8ae8-4885-8eb7-6a5221c41f80.mp4




</details>

**Review by ctkochan22 — DISMISSED** [review]

**Comment by ctkochan22** [comment]
@rushatgabhane we can't merge until we finish testing. What is the specific error you are getting in ONfido?

**Comment by ctkochan22** [comment]
> Not an error but Onfido is asking for a video verification of my face.

Can you just do the verification? You can't get past it?

**Comment by MariaHCD** [comment]
Thanks for testing @rushatgabhane! Let me see if I can reproduce the issue.

**Comment by MariaHCD** [comment]
Okay, I thought might've been due to the routing but both are the same on web and iOS:

Route on web:
```json
{
    "key": "ReimbursementAccount-y-7304S1Zt9qyPemSVhr_",
    "name": "ReimbursementAccount",
    "path": "/bank-account/",
    "params": {
        "stepToOpen": "BankAccountStep"
    }
}
```

Route on iOS:
```json
{
   "key":"ReimbursementAccount-5pIjiTW7wxxB8i2NwmGUP",
   "name":"ReimbursementAccount",
   "params":{
      "stepToOpen":"BankAccountStep"
   },
   "path":"/bank-account/"
}
```

But for some reason, the ContinueBankAccountSetup isn't showing on iOS. Will investigate further.

**Comment by MariaHCD** [comment]
Okay, I'm not sure why but the blank screen on iOS was due to wrapping ContinueBankAccountSetup in <View>:

Screenshot 2023-02-09 at 3 31 04 PM

Removed the <View> tags, re-tested and now it looks better:

https://user-images.githubusercontent.com/12268372/217801564-aa1fd62d-9e19-4970-8761-2b519afe6ed7.mp4

**Review by nkuoch — APPROVED** [review]

**Review by ctkochan22 — APPROVED** [review]
hmm strange. The view must have been a remnant when there were more components in there. 

@rushatgabhane if you can test today that would be amazing!

**Review by rushatgabhane — APPROVED** [review]
@ctkochan22  LGTM https://github.com/Expensify/App/pull/14726#pullrequestreview-1282900711

**Comment by MariaHCD** [comment]
Thanks, @rushatgabhane!

File: src/libs/actions/ReimbursementAccount/resetFreePlanBankAccount.js [hunk 4]
```diff
@@ -44,12 +44,31 @@ function resetFreePlanBankAccount(bankAccountID) {
                     key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
                     value: ReimbursementAccountProps.reimbursementAccountDefaultProps,
                 },
+                {
+                    onyxMethod: CONST.ONYX.METHOD.MERGE,
+                    key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
+                    value: {isLoading: true},
```
Comment (MariaHCD) [inline src/libs/actions/ReimbursementAccount/resetFreePlanBankAccount.js:50]: As a note for myself and reviewers, adding `isLoading` here caused this side effect: https://github.com/Expensify/App/issues/15114

https://github.com/Expensify/App/issues/15114#issuecomment-1430953467

**Pull request merged**